module/zfs/vdev_raidz.c

```diff
@@ -1811,10 +1812,12 @@ vdev_raidz_io_done_verified(zio_t *zio, raidz_row_t *rr)
 			vdev_t *vd = zio->io_vd;
 			vdev_t *cvd = vd->vdev_child[rc->rc_devidx];
 
-			if ((rc->rc_error == 0 || rc->rc_size == 0) &&
-			    (rc->rc_repair == 0)) {
+			if (!rc->rc_force_repair)
+				if (rc->rc_error == 0 || rc->rc_size == 0)
+					continue;
+
+			if (!rc->rc_allow_repair)
 				continue;
```

Comment:
I think this would be slightly clearer if you reversed the force/alloc repair checks. Do the rc_allow_repair check first and the rc_force_repair check afterwards:
```
    if (!rc->rc_allow_repair) {
        continue;
    } else if (!rc->rc_force_repair &&
      (rc->rc_error == 0 || rc->rc_size == 0) {
        continue;
    }
````

Reply:
Done!